### PR TITLE
adding nats cluster

### PR DESCRIPTION
**Pull request by tomusheq**

### CODE REVIEW

**Comment by adamw** [comment]
@tomusheq Is this configured with persistence & replication?

**Comment by adamw** [comment]
Any special requirements when using NATS? Why do we need an ELB here and not for other queues?

**Comment by wallyqs** [comment]
To get persistence (at least once delivery) would need to use either NATS Streaming (https://github.com/nats-io/nats-streaming-server) or JetStream (https://github.com/nats-io/jetstream).

**Comment by adamw** [comment]
@wallyqs Thanks for the pointers! I'm not (yet :) ) that proficient in the nats ecosystem, which one would you recommend to test when looking for a replicated, persistent, with at-least-once delivery message queue?

**Comment by wallyqs** [comment]
No problem :) For persistence and replication maybe the HA setup from this example could be an option, though it uses Raft for replication so would not be the best performance: https://docs.nats.io/nats-on-kubernetes/minimal-setup#ha-setup-using-statefulsets There is another deployment type called Fault Tolerance that uses a shared filesystem that could result in better perf in comparison but more involved to deploy.

**Comment by adamw** [comment]
@wallyqs Thanks again :) We are looking at deployments where the nodes are possibly independent, so we'll go with the raft option

**Comment by tomusheq** [comment]
> @tomusheq Is this configured with persistence & replication?

Yes I saw that NATS streaming is required, working on it.
Current config is without replication & persistence.

**Comment by adamw** [comment]
needs a merge/rebase :)

**Comment by adamw** [comment]
Latest seems to be 0.19.0? See https://github.com/nats-io/nats-streaming-server
Also maybe we could extract the version number to a configuration variable?

@tomusheq

**Comment by tomusheq** [comment]
> needs a merge/rebase :)

Done.

> Latest seems to be 0.19.0? See https://github.com/nats-io/nats-streaming-server
> Also maybe we could extract the version number to a configuration variable?
> 
I have added variable _nats_version_. Currently it is set to newest release available (v0.19.0).

File: ansible/roles/nats/templates/0nats-server.conf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    listen: 0.0.0.0:6222 
+    
+    routes = [
+      nats-route://{{ nats_host2 }}:6222
+      nats-route://{{ nats_host3 }}:6222
+    ]
+}
+
+streaming {
+  id: mqperf-cluster
```
Comment (adamw) [inline ansible/roles/nats/templates/0nats-server.conf:19]: hm I'm reading over the configuration, shouldn't this be `cluster_id`?  See https://docs.nats.io/nats-streaming-server/configuring/cfgfile - but maybe it works like this as well

**Pull request merged**